### PR TITLE
Add tests derived from RFC 3720, section B.4.

### PR DESCRIPTION
**Pull request by tseaver**

### CODE REVIEW

File: tests/test___init__.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import pytest
```
Comment (dhermes) [inline tests/test___init__.py]: License header needed

File: tests/test___init__.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+
+def _call_extend(crc, chunk):
+    import crc32c
```
Comment (dhermes) [inline tests/test___init__.py]: May as well just import at import time?

File: tests/test___init__.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    return crc32c.extend(crc, chunk, len(chunk))
+
+
+def test_extend_w_empty_chunk():
+    assert _call_extend(123, b'') == 123
+
+
+# From: https://tools.ietf.org/html/rfc3720#appendix-B.4
+iscsi_scsi_read_10_command_pdu = [
```
Comment (dhermes) [inline tests/test___init__.py]: PEP8, all caps var name?

**Review by dhermes — COMMENTED** [review]
1. Can you add a test that breaks `iscsi_scsi_read_10_command_pdu` into a few chunks and calls `extend()` multiple times, then compares against `value()` output?
1. Could you add a test to see how much `cffi` freaks out if the wrong types are passed in?

**Comment by dhermes** [comment]
@tseaver Is there a plan to do more re-factoring once these tests are in place?

**Comment by tseaver** [comment]
> Is there a plan to do more re-factoring once these tests are in place?

Yes, I plan to make the args explict for `value` and `extend`, and add in the hashlib-like wrapper.

**Comment by dhermes** [comment]
@tseaver I just noticed that AppVeyor decided to run for this (because it needed to build wheels).

Care to add a test run in AppVeyor? It should be easy since each environment already builds the wheels.

```
%PYTHON3X%/python \
  -m pip \
  install \
  --no-index \
  --find-links . \
  py_crc32c
```

File: tests/test___init__.py (hunk 1 above)
Comment (tseaver) [reply tests/test___init__.py]: e8bbd2b

File: tests/test___init__.py (hunk 2 above)
Comment (tseaver) [reply tests/test___init__.py]: e8bbd2b

File: tests/test___init__.py (hunk 3 above)
Comment (tseaver) [reply tests/test___init__.py]: e8bbd2b

**Comment by tseaver** [comment]
> Can you add a test that breaks iscsi_scsi_read_10_command_pdu into a few chunks and calls extend() multiple times, then compares against value() output?

Added in e8bbd2b

File: README.md [hunk 4]
```diff
@@ -100,3 +100,28 @@ On Windows: TBD.
...
+$ cd crc32c/build
+$ /path/to/venv/bin/cmake \
+>   -DCRC32C_BUILD_TESTS=no \
+>   -DCRC32C_BUILD_BENCHMARKS=no \
+>   -DBUILD_SHARED_LIBS=yes \
+>   -DCMAKE_INSTALL_PREFIX:PATH=${CRC32C_INSTALL_PREFIX} \
+>   ..
+$ make all install
+$ cd ../..
+$ python3.Y -m venv usr
```
Comment (dhermes) [inline README.md]: 1. Is `venv` preferred over `virtualenv`? (I was under the impression that `venv` was not well-liked)
1. Can you name it something other than `usr`, like `venv`?

File: src/crc32c_build.py [hunk 5]
```diff
@@ -23,9 +23,11 @@ def get_kwargs():
     library_dirs = [os.path.join(install_prefix, "lib")]
     if os.name == "nt":
         library_dirs.append(os.path.join(install_prefix, "bin"))
+    rpath = os.pathsep.join(library_dirs)
     return {
         "library_dirs": library_dirs,
         "include_dirs": [os.path.join(install_prefix, "include")],
+        "extra_link_args": ["-Wl,-rpath={}".format(rpath)],
```
Comment (dhermes) [inline src/crc32c_build.py:30]: We should verify that this doesn't "break" the `manylinux` wheels? I can do the same for OS X?

**Comment by tseaver** [comment]
> Could you add a test to see how much cffi freaks out if the wrong types are passed in?

I don't know how useful a test would be for asserting this traceback:

```python
>>> import crc32c
>>> crc32c.value('', 0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/tseaver/projects/agendaless/Google/src/py-crc32c/src/crc32c/__init__.py", line 24, in value
    return crc32c._crc32c_cffi.lib.crc32c_value(*args, **kwargs)
TypeError: initializer for ctype 'uint8_t *' must be a bytes or list or tuple, not str
```

File: tests/test___init__.py [hunk 6]
```diff
@@ -1,38 +1,55 @@
...
-    (bytes(iscsi_scsi_read_10_command_pdu), 0xd9963a56),
+    (bytes(ISCSI_SCSI_READ_10_COMMAND_PDU), ISCSI_CRC),
 ]
 
 
-def _call_value(chunk):
-    import crc32c
-    return crc32c.value(chunk, len(chunk))
+def test_extend_w_empty_chunk():
+    crc = 0
```
Comment (dhermes) [inline tests/test___init__.py]: Unused variable here

**Review by dhermes — APPROVED** [review]
LGTM other than a few nits

**Comment by dhermes** [comment]
```
TypeError: initializer for ctype 'uint8_t *' must be a bytes or list or tuple, not str
```

is a very helpful error message though!

File: README.md (hunk 4 above)
Comment (tseaver) [reply README.md]: > 1. Is venv preferred over virtualenv? (I was under the impression that venv was not well-liked)

`venv` is present only on Python >= 3.3, but it is ubiquitous there (don't have to bootstrap it).

> 2. Can you name it something other than usr, like venv?

OK.  I figured reusing `usr` made sense (share the `bin` and `lib` directories already there after the `make` step above.

File: README.md [hunk 7]
```diff
@@ -100,3 +100,44 @@ On Windows: TBD.
...
+
+Now, run the tests:
+
+```bash
+$ venv/bin/pip install pytest
+$ venv/bin/pip install -e .
+$ venv/bin/py.test tests/
+============================= test session starts ==============================
+platform linux -- Python 3.6.7, pytest-3.10.0, py-1.7.0, pluggy-0.8.0
+rootdir: /home/tseaver/projects/agendaless/Google/src/py-crc32c, inifile:
```
Comment (dhermes) [inline README.md:137]: @tseaver Path snuck in here

Comment (dhermes) [reply README.md:137]: @tseaver Can you remove this in #4?

Comment (tseaver) [reply README.md:137]: 43207e0

**Pull request merged**